src/main/java/com/dinegood/springbootproject/dao/CategoryDao.java

```
@@ -0,0 +1,30 @@
+package com.dinegood.springbootproject.dao;
+
+import com.dinegood.springbootproject.model.Categorydetails;
+import com.dinegood.springbootproject.repo.Repo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+@Component
+public class CategoryDao {
+    @Autowired
+    private Repo repo;
+    public List<Categorydetails> getcatgs(){
+        try{List<Categorydetails> category= (List<Categorydetails>) this.repo.findAll();
+        return category;}
+        catch (Exception e){
+            e.printStackTrace();
+        }
+        return null;
```

For both methods in this class
1. return result inline.
2. Instead of returning null, throw exception to parent class